p4proto/kctrl/switchapi/README.md

```diff
@@ -0,0 +1,165 @@
+<!--
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+- -->
+
+# SwitchAPI library
+
+## Table of contents
+1. [Overview](#overview)
+2. [FDB](#fdb)
+3. [RIF](#rif)
+4. [VRF](#vrf)
+5. [RMAC](#rmac)
+6. [Neighbor](#neighbor)
+7. [Nexthop](#nexthop)
+8. [Route](#route)
+9. [Platform Dependent API](#platform_dependent_api)
+
+
+## Overview <a name="overview"></a>
+The Switchapi library is an abstraction layer that exposes higher level API's used to interact with target backend to program the data plane described in the linux_networking.p4.
+Modules supported in switchapi are FDB, Nexthop, Neighbor, ECMP, Route, VRF, RMAC, Tunnel.
+
+For each module, as part of initialization, device context is set, hashtable is initialized and a unique handle is allocated and initialized based on the type of module.
+
+## FDB <a name="fdb"></a>
+ * During initialization, device context is set, hashtable is initialized and handle is allocated for both Tx (SWITCH_HANDLE_TYPE_L2_FWD_TX) and Rx (SWITCH_HANDLE_TYPE_L2_FWD_RX) and updated in device context
```

Comment:
somehow these are shown in a different way while we view it on browser.
https://github.com/ipdk-io/ovs/blob/linux_networking_documentation/p4proto/kctrl/switchapi/README.md#fdb-